components/ntp_background_images/browser/view_counter_model.cc

```diff
@@ -25,14 +25,16 @@ bool ViewCounterModel::ShouldShowBrandedWallpaper() const {
 
 void ViewCounterModel::ResetCurrentWallpaperImageIndex() {
   current_wallpaper_image_index_ = 0;
+  current_branded_wallpaper_image_index_ = 0;
 }
 
 void ViewCounterModel::RegisterPageView() {
+  // Do not add  DCHECK_NE for total_background_image_count_ here. RegisterPageView can happen before BI finished downloading.
   DCHECK_NE(-1, total_image_count_);
```

Comment:
NTP SI could be same situation on android
This can be called before NTP SI is ready. but this doesn't happen because RegisterPageView() is only called after getting OnComponentReady?